From 2bdc308e19e1ec54974b3fdc5d961dd71fab8027 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Fri, 4 Feb 2011 12:20:03 -0200
Subject: [RHEL6 qemu-kvm PATCH 27/27] Drop qemu_mutex_iothread during migration

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <8134328e4eafcb47a4f3e89f401596b02256d075.1296800500.git.quintela@redhat.com>
Patchwork-id: 17742
O-Subject: [PATCH 8/8] Drop qemu_mutex_iothread during migration
Bugzilla: 643970
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

This way we don't delay other vcpus with migration, only iothread.

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 vl.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 vl.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/vl.c b/vl.c
index d899c9c..2c85070 100644
--- a/vl.c
+++ b/vl.c
@@ -3379,6 +3379,7 @@ static int ram_save_live(Monitor *mon, QEMUFile *f, int stage, void *opaque)
     bytes_transferred_last = bytes_transferred;
     t0 = get_clock();
 
+    qemu_mutex_unlock_iothread();
     i = 0;
     while (!qemu_file_rate_limit(f)) {
         int ret;
@@ -3400,7 +3401,7 @@ static int ram_save_live(Monitor *mon, QEMUFile *f, int stage, void *opaque)
         }
         i++;
     }
-
+    qemu_mutex_lock_iothread();
     t0 = get_clock() - t0;
     bwidth = (bytes_transferred - bytes_transferred_last) / t0;
 
-- 
1.7.3.2

